kafka.go

```diff
@@ -0,0 +1,390 @@
+		if err != nil {
+			logrus.Error("Unable to create topic in kafka : ", err)
+			return
+		}
+
+		// listening a topic
+		go func() {
+			consumer, err := client.consumer.ConsumePartition(publishBuilder.topic, 0, sarama.OffsetNewest)
+			if err != nil {
+				logrus.Error("Unable to consume topic from kafka : ", err)
+				return
+			}
+
+			for timeout := time.After(time.Duration(publishResponseTimeout) * time.Millisecond); ; {
+				select {
+				case consumerMessage := <-consumer.Messages():
+					if !processResponseMessage(consumerMessage, publishBuilder.messageId) {
+						continue
+					}
+					break
+				case err = <-consumer.Errors():
+					callback := publishBuilder.callback
+					callback(nil, err)
+					break
+				case <-timeout:
+					callback := publishBuilder.callback
+					callback(nil, rtoError)
+					break
+				}
+
+			}
+		}()
+	}
+
+	// send message
+	_, _, err := client.syncProducer.SendMessage(&sarama.ProducerMessage{
+		Timestamp: time.Now().UTC(),
+		Value: &PublishBuilderEntry{
+			PublishBuilder: *publishBuilder,
+		},
+		Topic: publishBuilder.topic,
+	})
+	if err != nil {
+		logrus.Error("Unable to publish message in publish with response to kafka : ", err)
+		return
+	}
+}
+
+// Register add subscriber for a topic and register callback function
+func (client *KafkaClient) Register(subscribeBuilder *SubscribeBuilder) {
+
+	if registerCallback(subscribeBuilder.topic, subscribeBuilder.messageType, subscribeBuilder.callback) {
+		logrus.Error("Topic and Message Type already registered")
+		return
+	}
+
+	err := createTopic(client.broker, subscribeBuilder.topic)
+	if err != nil {
+		logrus.Error("Unable to create topic in kafka : ", err)
+		return
+	}
+
+	// listening a topic
+	// each topic would have his own goroutine
+	go func() {
+		consumer, err := client.consumer.ConsumePartition(subscribeBuilder.topic, 0, sarama.OffsetNewest)
+		if err != nil {
+			logrus.Error("Unable to consume topic from kafka : ", err)
+			return
+		}
+
+		for {
+			select {
+			case consumerMessage := <-consumer.Messages():
+				processMessage(consumerMessage)
+				break
+			case err = <-consumer.Errors():
+				callback := subscribeBuilder.callback
+				callback(nil, err)
+				break
+			}
+
+		}
+	}()
+}
+
+// createTopic create kafka topic
+func createTopic(broker *sarama.Broker, topicName string) error {
+	topicDetail := &sarama.TopicDetail{}
+	topicDetail.NumPartitions = int32(1)
+	topicDetail.ReplicationFactor = int16(1)
+	topicDetail.ConfigEntries = make(map[string]*string)
+
+	topicDetails := make(map[string]*sarama.TopicDetail)
+	topicDetails[topicName] = topicDetail
+	request := sarama.CreateTopicsRequest{
+		Timeout:      time.Second * 15,
+		TopicDetails: topicDetails,
+	}
+
+	_, err := broker.CreateTopics(&request)
+	return err
+}
+
+// deleteTopic delete kafka topic
+func deleteTopic(broker *sarama.Broker, topicName string) error {
+	request := sarama.DeleteTopicsRequest{
+		Timeout: time.Second * 15,
```

Comment:
also this one, better to put in a constant

Reply:
sure, I will update it
thanks